src/Test/StateMachine/Internal/Sequential.hs

```diff
@@ -255,7 +255,8 @@ checkSequentialInvariant
     monitor $ label s
     pre $ precondition m cmd
     resp <- run $ liftSem (sem m miref) cmd miref
+    let m' = transition m cmd resp
     liftProperty $
-      counterexample ("\nThe post-condition for `" ++ s ++ "' failed!\n\n" ++ show m) $
+      counterexample ("\nThe post-condition for `" ++ s ++ "' failed!\n\n" ++ show m') $
```

Comment:
I basically just did it this way to get the desired output for the die hard example.

You reckon we should print both? Something like: "The model was `m`, then `s` happend, post-condition for `s` doesn't hold in the new model `m'`?